Feat/add testsuite

Candy9797 opened the pull request.

Candy9797 committed:
feat: add testsuite

Candy9797 committed:
ci: trigger test workflow on any branch

Candy9797 committed:
chore: remove optional documentation files to keep PR focused

Candy9797 committed:
docs: translate testsuite README to English

Candy9797 committed:
docs: remove testsuite README to keep PR minimal

Candy9797 committed:
fix: ensure reliable PostgreSQL authentication in container mode

- Explicitly set username=postgres and password=my-secret
- Add wait strategy for container readiness
- Ensures consistent authentication across all environments

Candy9797 committed:
chore: remove Chinese comment, use English for open source

Candy9797 committed:
docs: add comment explaining authentication configuration

Stumble commented on .github/workflows/test-with-containers.yml:
```diff
@@ -0,0 +1,34 @@
+  push:
+    branches: [ "**" ]
+  pull_request:
+    branches: [ "**" ]
```
need stricter condition. Not all branches need to be CI tested.

Candy9797 replied:
stricter condition： 
on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: 
      - "main"
      - "feat/**"
      - "fix/**"

Stumble commented on .github/workflows/test-with-containers.yml:
```diff
@@ -0,0 +1,34 @@
+      run: make test-container
+      env:
+        # Testcontainers will automatically start PostgreSQL containers
+        WPGX_TEST_USE_CONTAINER: "true"
```
better naming: USE_TEST_CONTAINERS

Candy9797 replied:
已全部替换成USE_TEST_CONTAINERS哈

Stumble commented on examples/container_test_example.go:
```diff
@@ -0,0 +1,86 @@
+// - true: 使用 testcontainers 自动启动 PostgreSQL 容器
+// - false 或未设置: 使用直连模式（需要预先启动 PostgreSQL）
+func NewExampleTestSuite() *ExampleTestSuite {
+	return &ExampleTestSuite{
```
No chinese comments

Candy9797 replied:
已修改～

Stumble commented on examples/container_test_example.go:
```diff
@@ -0,0 +1,86 @@
+	"github.com/stretchr/testify/suite"
+
+	sqlsuite "github.com/stumble/wpgx/testsuite"
+)
```
don't alias to sqlsuite

Candy9797 replied:
Modified~

Stumble commented on testsuite/testsuite.go:
```diff
@@ -76,6 +86,71 @@ func (suite *WPgxTestSuite) GetPool() *wpgx.Pool {
+	// Start PostgreSQL container
+	// Use postgres superuser for reliable authentication across all environments.
+	// The default testcontainers credentials (test/test) can have authentication issues.
+	container, err := postgres.Run(ctx,
+		"postgres:14.5",
+		postgres.WithDatabase(suite.Testdb),
+		postgres.WithUsername("postgres"),
+		postgres.WithPassword("my-secret"),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("database system is ready to accept connections").
+				WithOccurrence(2).
+				WithStartupTimeout(60*time.Second)),
+	)
```
change them to constants.

Candy9797 replied:
Modified~

Candy9797 committed:
ci: restrict workflow to main branch only

Candy9797 committed:
refactor: rename env var to USE_TEST_CONTAINERS for better clarity

- Changed from WPGX_TEST_USE_CONTAINER to USE_TEST_CONTAINERS
- Updated all files: testsuite.go, Makefile, README.md, examples, workflow
- More concise and follows common naming conventions

Candy9797 committed:
docs: translate Chinese comments to English and add fix/** branch

- Translate all Chinese comments in examples to English
- Add fix/** branch to PR workflow triggers
- Ensures code is fully English for international contributors

Candy9797 committed:
refactor: remove import alias and extract constants

- Remove sqlsuite alias, use testsuite package name directly
- Extract container configuration values to constants (image, user, password, timeout)
